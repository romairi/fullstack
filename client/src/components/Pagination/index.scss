.pagination {

  .pagination_page_number {
    user-select: none;
  }
}
