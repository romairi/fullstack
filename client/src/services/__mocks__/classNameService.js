module.exports = {
    buildClassName: jest.fn(() => 'COMPUTED_CLASS_NAME'),
};
