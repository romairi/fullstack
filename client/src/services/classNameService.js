export const buildClassName = classes => classes.filter(c => !!c).join(' ');
