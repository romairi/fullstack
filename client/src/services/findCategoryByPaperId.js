export function findCategoryByPaperId(categories, paperId) {
    return categories.find(c => c.paperItems.find(paper => paper.paperId === paperId));
}
