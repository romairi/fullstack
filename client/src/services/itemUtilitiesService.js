export function searchByFields(items, field) {
    return items.filter(item => [item.title, item.summary, ...item.authors.filter(a => Array.isArray(a) && a.length > 0).map(a => a[0])]
        .map(text => text.toLowerCase())
        .filter(text => text.includes(field.toLowerCase()))
        .length > 0
    );
}
