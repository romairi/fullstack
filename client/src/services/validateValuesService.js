export function reformatErrors(details) {
    return Array.isArray(details) ? details
        .filter(item => !!item?.context?.key && !!item.message)
        .reduce((acc, cur) => {
            acc[cur.context.key] = cur.message;
            return acc;
        }, {}) : {};
}

export function validateValues(values, schema) {
    const results = schema.validate(values, {abortEarly: false});
    let errors = {};

    if (results.error) {
        errors = reformatErrors(results.error.details);
    }
    return errors;
}
