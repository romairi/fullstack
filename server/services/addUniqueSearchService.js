function formatSearchTags(listTags) {
    return listTags.map(tag => tag.toLowerCase()).sort().join();
}

function uniqueSearch(userSearches, includeTags, excludeTags) {
    const includeList = [];
    const excludeList = [];
    let i = 0;

    while (userSearches.length > i) {
        includeList.push(formatSearchTags(userSearches[i].include_tags));
        excludeList.push(formatSearchTags(userSearches[i].exclude_tags));
        i++;
    }

    const searchIncTags = formatSearchTags(includeTags.map(item => item));
    const searchExcTags = formatSearchTags(excludeTags.map(item => item));
    const foundInclude = includeList.find(item => item === searchIncTags);
    const foundExclude = excludeList.find(item => item === searchExcTags);
    const includeTagExists = searchIncTags.includes(foundInclude);
    const excludeTagExists = searchExcTags.includes(foundExclude);

    return (includeTagExists && excludeTagExists);
}

module.exports = {uniqueSearch};
