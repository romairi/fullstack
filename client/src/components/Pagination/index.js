import React from 'react';
import "./index.scss";
import Button from "@material-ui/core/Button";
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
import ChevronRightIcon from '@material-ui/icons/ChevronRight';
import {buildClassName} from "../../services/classNameService";


function Pagination({page, className, onChangePage, isLastPage}) {
    return (
        <div className={buildClassName(["pagination", className])}>
            <Button className={buildClassName(["pagination_button", "button_left", (page === 0) && "disable"])}
                    disabled={page === 0} onClick={() => onChangePage(page - 1)}>
                <ChevronLeftIcon fontSize="large"/>
            </Button>
            <span className="pagination_page_number">{page}</span>
            <Button className={buildClassName(["pagination_button", "button_right", isLastPage && "disable"])}
                    disabled={isLastPage} onClick={() => onChangePage(page + 1)}>
                <ChevronRightIcon fontSize="large"/>
            </Button>
        </div>
    );
}

export default Pagination;
