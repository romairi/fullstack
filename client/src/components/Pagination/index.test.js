import React from 'react';
import renderer from 'react-test-renderer';
import Pagination from './index';
import {buildClassName} from "../../services/classNameService";

jest.mock('../../services/classNameService');
jest.mock('@material-ui/core/Button', () => () => <div className='Button'/>);
jest.mock('@material-ui/icons/ChevronLeft', () => () => <div className='ChevronLeft'/>);
jest.mock('@material-ui/icons/ChevronRight', () => () => <div className='ChevronRight'/>);

describe('Pagination', () => {
    describe('render', () => {
        it('should render the component', () => {
            const component = renderer.create(
                <Pagination page={0} isLastPage={10} onChangePage={0}/>
            );

            expect(component.toJSON()).toMatchSnapshot();
            expect(buildClassName.mock.calls.pop()).toMatchSnapshot();
        });
    });
});
