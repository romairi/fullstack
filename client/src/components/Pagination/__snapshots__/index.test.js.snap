// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination render should render the component 1`] = `
<div
  className="COMPUTED_CLASS_NAME"
>
  <div
    className="Button"
  />
  <span
    className="pagination_page_number"
  >
    0
  </span>
  <div
    className="Button"
  />
</div>
`;

exports[`Pagination render should render the component 2`] = `
Array [
  Array [
    "pagination_button",
    "button_right",
    "disable",
  ],
]
`;
